errors.go

```diff
@@ -150,6 +172,17 @@ func (e *contextAlreadyDoneError) Unwrap() error {
 	return e.err
 }
 
+// newContextAlreadyDoneError wraps a context error in `contextAlreadyDoneError`. If the context was cancelled or its
+// deadline passed, the returned error is also wrapped by `ErrTimeout`.
+func newContextAlreadyDoneError(ctx context.Context) (err error) {
+	ctxErr := ctx.Err()
+	err = &contextAlreadyDoneError{err: ctxErr}
+	if ctxErr != nil {
+		err = &ErrTimeout{err: err}
+	}
+	return err
+}
+
```

Comment:
ctxErr can never be nil. This function should always do the double wrapping.

Or... this function should be removed and we go back to directly creating `contextAlreadyDoneError`s and `pgconn.Timeout` should also check for `contextAlreadyDoneError`. An extra check there vs an extra allocation here. 🤷

Reply:
I've modified this function to always double-wrap the error: 1134028

Regarding the change to `pgconn.Timeout`, see [my comment here](https://github.com/jackc/pgconn/pull/79#discussion_r664851620)